app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
```

Comment:
Nitpick: Let's use `find_by!` instead of `find_by` so that we fail with a clearer error message if we can't find it?